### PR TITLE
Add content restore dialog

### PR DESCRIPTION
**Pull request by dcsaszar**
The other end of #218 

The menu item can be activated by https://edit.scrivito.com/http://localhost:8080/?restoreContent

### CODE REVIEW

**Comment by apepper** [comment]
FYI: I wait for the refactoring to `unstable_JrRestApi` before reviewing this PR.

**Review by apepper — CHANGES_REQUESTED** [review]
FIRST-GLIMPSE-REVIEW ([65284ce](https://github.com/Scrivito/scrivito-portal-app/pull/222/commits/65284ce951821c7f3276bf732b236f9b30e5a563))

Nice start 🚀!

Please see my comments.

File: src/Extensions/RestoreContent/restoreContent.ts [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        )}`,
+      )}`,
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: 'grant_type=client_credentials',
+  })
+
+  if (response.status !== 200) throw new Error('Failed to get API access.')
+
+  return (await response.json()).access_token
```
[the remark this answers is not in the document]
Comment (dcsaszar) [reply src/Extensions/RestoreContent/restoreContent.ts:196]: When we fetch the `iam/instances/${import.meta.env.SCRIVITO_TENANT}/userinfo` or `iam/instances/${import.meta.env.SCRIVITO_TENANT}/userinfo?service=https%3A%2F%2Fapi.scrivito.com` endpoint with `unstable_JrRestApi.fetch` or standard `fetch` with `currentEditor().authToken()` we always receive empty `instance_permissions`, which indicates missing Scrivito API access.

Maybe this is caused by a missing or incompatible (`!== "https://api.scrivito.com"`) `audience` to satisfy 

> which backend service(s) this token is intended for

https://docs.google.com/document/d/1MoJ1LlyxD0sZCUpzuWB2t2LaShSbwHlO2rBOOXzMbbU/edit#heading=h.bholi5gfql9y 

https://github.com/search?q=repo%3Ainfopark%2Fscrivito_js%20audience%3A&type=code

File: src/Extensions/RestoreContent/RestoreContentEditingConfig.ts [hunk 2]
```diff
@@ -5,7 +5,6 @@ extendMenu((menu) => {
 
   menu.insert({
     enabled: canWrite(),
-    icon: 'https://cdn.jsdelivr.net/npm/bootstrap-icons/icons/stars.svg',
```
Comment (apepper) [inline src/Extensions/RestoreContent/RestoreContentEditingConfig.ts]: `bootstrap-icons` is now available as a package in `main`. So you could import `stars.svg` from there.

File: src/Extensions/index.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import.meta.glob(['./**/*.ts', './**/*.tsx'], { eager: true })
```
Comment (apepper) [inline src/Extensions/index.ts:1]: If I'm not mistaken, the consequence of this, is that _all_ `contentDump` is loaded on every editor visit of the page. `contentDump` should only be loaded, once the restore process is started.

Comment (dcsaszar) [reply src/Extensions/index.ts:1]: The recursive imports only create maps of filename to promise that load the content dump items on demand (after the dialog opens).
However, I'll make the top-level import more specific.

**Comment by apepper** [comment]
I took your changes and pushed it to `demo2_content`. A preview deployment is available at https://edit.scrivito.com/b74df6074bfe11bf857a233c4be21103~https://demo2-content.scrivito-portal-app.pages.dev/b74df6074bfe11bf857a233c4be21103?restoreContent .

But restore fails on it:

```
🔍 Looking for existing content
🗑️ Deleting existing content
🐣 Restoring content
❌ Error: TypeError: Failed to fetch dynamically imported module: https://demo2-content.scrivito-portal-app.pages.dev/assets/060918d8e44ae54f%2F1dcf5d167865%2FAdobeStock_292341430-ZHT4PqlZ.js
⏩ 3 of 110 uploading AdobeStock_292341430.jpeg
```

Please look into this.

**Comment by dcsaszar** [comment]
@apepper REVIEW 🎉 🙈

**Comment by apepper** [comment]
Will be implemented differently. Thanks anyway for the great work!

**Pull request closed without merging**